zebra-consensus/src/transaction/check.rs

```diff
@@ -30,10 +32,21 @@ pub fn has_inputs_and_outputs(tx: &Transaction) -> Result<(), TransactionError>
     let n_spends_sapling = tx.sapling_spends_per_anchor().count();
     let n_outputs_sapling = tx.sapling_outputs().count();
     let n_actions_orchard = tx.orchard_actions().count();
+    let flags_orchard = tx.orchard_flags().unwrap_or_else(Flags::empty);
 
-    if tx_in_count + n_spends_sapling + n_joinsplit + n_actions_orchard == 0 {
+    if tx_in_count
+        + n_spends_sapling
+        + n_joinsplit
+        + (n_actions_orchard > 0 && flags_orchard.contains(Flags::ENABLE_SPENDS)) as usize
```

Comment:
I don't think there can be an arithmetic overflow in practice, but still, it's not expressing the rule in the same way as the spec.

Reply:
Thanks, i created a ticket to improve this code https://github.com/ZcashFoundation/zebra/issues/2410